src/EditorPanel.tsx

```diff
@@ -432,7 +432,7 @@ export const EditorPanel: React.FC<{
             {/* This extra div acts as a placeholder that is supposed to stretch while EditorWithXStateImports lazy-loads (thanks to `1fr` on the grid) */}
             <div>
               <EditorWithXStateImports
-                defaultValue={defaultValue}
+                value={value}
```

Comment:
Silly question, but will this cause more rerenders (since `value` is now making a round-trip with the machine)?

Reply:
No, I don't think so - value has been making a round trip consistently before, so I don't think that the performance drop-off is noticeable.

I thought about setting the value programmatically, but it requires lifting the editor ref up, which feels uglier to me.

Reply:
For reference - the `Editor` has to compare the old and new values to skip over forcefully executing edits:
https://github.com/suren-atoyan/monaco-react/blob/5a308001a49a10001b6d7c63dcb7be62fcf7676b/src/Editor/Editor.js#L84-L92

Even though this is a basic string comparison, the text itself can be lengthy and this will be executed on each keystroke. I've compared the responsiveness of the editor on the production and on this branch and I can feel a little bit of a slowdown, the input sometimes lags a little bit (not everyone has M1 Matt! 😉 ). However, I've also rechecked how the `dev` branch performs locally and I can see a similar slowdown there. Therefore this might be more of a React issue than those linked lines of code - sometimes the perf can be quite different between production and development builds of React.

To sum it up - I wouldn't bother with this and just observe if this comes with any production perf regressions.